src/hmem.c

```diff
@@ -126,3 +139,29 @@ ssize_t ofi_copy_to_hmem_iov(const struct iovec *hmem_iov,
 				     hmem_iov_offset, src, size,
 				     OFI_COPY_BUF_TO_IOV);
 }
+
+int ofi_get_hmem_iface(const void *buf, enum fi_hmem_iface *iface)
+{
+	int i;
+	int ret;
+
+	/* Search backwards since FI_HMEM_SYSTEM will claim the buffer. */
+	for (i = ARRAY_SIZE(hmem_ops) - 1; i >= FI_HMEM_SYSTEM; i--) {
```

Comment:
If we make the is_hmem_iface call return a bool, this function could be updated to return the fi_hmem_iface directly.  I haven't looked ahead in the patches to see what the code does if this function actually returns an error.

In any case, we don't need to iterate over FI_HMEM_SYSTEM and can just return that directly if the other hmem checks fail.

Reply:
Will do.